### PR TITLE
Upgrade gomock

### PR DESCRIPTION
**Pull request by cjellick**
Doing so breaks a couple tests, so this PR fixes those as well.

Signed-off-by: Craig Jellick <craig@acorn.io>

### Checklist
- [x] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [ ] The title of this PR ends with a link to the main issue being address in paranthesis, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/acorn/pull/1199)
- [x] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keyworkds](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [x] Commits follow [contributing guidance](https://github.com/acorn-io/acorn/blob/main/CONTRIBUTING.md#commits)
- [x] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [x] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

File: pkg/client/info_test.go [hunk 1]
```diff
@@ -102,12 +103,12 @@ func TestMultiClientInfo(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	mFactory := mocks.NewMockProjectClientFactory(ctrl)
 	mFactory.EXPECT().List(gomock.Any()).Return([]client.Client{&defaultClient1, &defaultClient2}, nil)
-	mFactory.EXPECT().DefaultProject().Return("test1")
+	//mFactory.EXPECT().DefaultProject().Return("test1")
```
Comment (StrongMonkey) [inline pkg/client/info_test.go:106]: Any reason to comment out instead of removing?

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -75,8 +75,6 @@ func TestAcornDNSStates(t *testing.T) {
 			},
 			expectedClusterDomains: []string{".custom.com"},
 			prepare: func(f *mocks.MockReader) {
-				f.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
-				f.EXPECT().List(gomock.Any(), gomock.Any()).Return(nil)
 			},
```
Comment (njhale) [inline pkg/config/config_test.go:78]: nit: seems like the logic below can handle a `nil` `prepare`, so this empty literal can be dropped entirely.

File: pkg/client/info_test.go (hunk 1 above)
Comment (cjellick) [reply pkg/client/info_test.go:106]: oh no! was testing the fix

Comment (cjellick) [reply pkg/client/info_test.go:106]: will delete

**Review by jsilverio22 — APPROVED** [review]
lgtm when tests pass

**Review by StrongMonkey — APPROVED** [review]

**Review by tylerslaton — APPROVED** [review]

**Review by njhale — APPROVED** [review]
just a nit, so LGTM

File: pkg/config/config_test.go (hunk 2 above)
Comment (cjellick) [reply pkg/config/config_test.go:78]: shoot. good call. ugh, rushing this PR and it shows. 🤦

**Pull request merged**